R/impact_central.R

```diff
@@ -140,27 +144,30 @@ impact_by_year_of_vaccination <- function(meta1, raw_impact, fvps, fvps_updates
 
   if (method == "method2a"){
     if (vaccine_delivery$activity_type[1L] == "routine"){
-      raw_impact <- raw_impact[raw_impact$time %in% cohorts_vaccinated, ]
+      raw_impact <- raw_impact[raw_impact$time %in% (vaccination_years - min(fvps$age)), ]
     } else {
       raw_impact <- raw_impact[raw_impact$time >= min(vaccination_years), ]
     }
     tot_impact <- stats::aggregate(value ~ country + burden_outcome, raw_impact, sum, na.rm = TRUE)
     tot_fvps <- stats::aggregate(fvps~country, fvps, sum, na.rm = TRUE)
     d <- merge_by_common_cols(tot_impact, tot_fvps, all = TRUE)
   } else if (method == "method2b"){
-    cohort_impact <- raw_impact[raw_impact$time %in% cohorts_vaccinated, c("country", "time", "burden_outcome", "value")]
-    cohort_fvps <- stats::aggregate(fvps ~ country + time, fvps, sum, na.rm = TRUE)
+    cohort_impact <- raw_impact[raw_impact$time %in% (min(fvps$time):max(fvps$time)), c("country", "time", "burden_outcome", "value")]
+    cohort_fvps <- aggregate(fvps ~ country + time, fvps, sum, na.rm = TRUE)
```

Comment:
Removing the `stats::` part will trigger a note or warning in R CMD check - it's good to try and remove these just to reduce noise. Unless we want to use aggregate from one of the other packages - in that case we should include the correct package.

Reply:
I see. I will add stats:: back to avoid warnings.